exe/roo_on_rails

```diff
@@ -3,12 +3,103 @@
 require 'rubygems'
 require 'roo_on_rails/config'
 
-if ARGV.include? 'sidekiq'
-  require 'roo_on_rails/sidekiq/loader'
-  RooOnRails::Sidekiq::Loader.run
-else
-  require 'roo_on_rails/harness'
-  module RooOnRails
-    Harness.new(try_fix: true, context: Config.load).run
+module RooOnRails
+  module CLI
+    class Base
+      attr_reader :argv
+
+      def initialize(argv)
+        @argv = argv
+      end
+
+      def usage
+        raise NotImplementedError
+      end
+
+      protected
+
+      def _log(message)
+        $stderr.puts message
+      end
+
+      def _assert_argv(enum)
+        return if enum.include?(argv.length)
+        $stderr.puts enum.inspect
+        $stderr.puts argv.inspect
+        _log 'Wrong number of arguments.'
+        _log "Usage: #{usage}"
+        exit 1
+      end
+    end
+
+    class Sidekiq < Base
+      def usage
+        'roo_on_rails sidekiq'
+      end
+
+      def run
+        _assert_argv(0..0)
+        require 'roo_on_rails/sidekiq/loader'
+        RooOnRails::Sidekiq::Loader.run
+      end
+    end
+
+    class Harness < Base
+      def usage
+        'roo_on_rails harness [-f|--fix]'
+      end
+
+      def run
+        _parser.parse!(argv)
+        _assert_argv(0..0)
+        require 'roo_on_rails/harness'
+        RooOnRails::Harness.new(try_fix: _options[:fix], context: Config.load).run
+      end
+
+      private
+
+      def _options
+        @options ||= { fix: false, env: nil }
+      end
+
+      def _parser
+        require 'optionparser'
+
+        OptionParser.new do |o|
+          o.banner = usage
+
+          o.on('-f', '--fix', 'Attempt to fix failed checks') do
+            _options[:fix] = true
+          end
+
+          o.on('-e', '--environment', 'Application environments to check') do |v|
```

Comment:
Should this be `--env` according to the README? Also it's not clear to me how this would work if I wanted to check multiple environments? If I did `--env staging --env production` then this would only check `production` because it would overwrite `_options[:env]` the second time?